include/deal.II/lac/la_vector.h

```diff
@@ -0,0 +1,440 @@
+     * Copy constructor. Sets the dimension to that of the given vector and
+     * copies all elements.
+     */
+    Vector(const Vector<Number> &V);
+
+    /**
+     * Constructor. Set dimension to @p n and initialize all elements with
+     * zero.
+     *
+     * The constructor is made explicit to avoid accident like this:
+     * <tt>v=0;</tt>. Presumably, the user wants to set every element of the
+     * vector to zero, but instead, what happens is this call:
+     * <tt>v=Vector@<Number@>(0);</tt>, i.e. the vector is replaced by one of
+     * length zero.
+     */
+    explicit Vector(const size_type n);
+
+    /**
+     * Initialize the vector with a given range of values pointed to by the
+     * iterators. This function exists in analogy to the @p std::vector class.
+     */
+    template <typename InputIterator>
+    Vector(const InputIterator first, const InputIterator last);
+
+    /**
+     * Destructor, deallocates memory.
+     */
+    virtual ~Vector();
+
+    /**
+     * Copies the data of the input vector @p in_vector.
+     */
+    Vector<Number> &operator= (const Vector<Number> &in_vector);
+
+    /**
+     * Copies the data of the input vector @p in_vector.
+     */
+    template <typename Number2>
+    Vector<Number> &operator= (const Vector<Number2> &in_vector);
+
+    /**
+     * Sets all elements of the vector to the scalar @p s. This operation is
+     * only allowed if @p s is equal to zero.
+     */
+    Vector<Number> &operator= (const Number s);
+
+    /**
+     * Multiply the entire vector by a fixed factor.
+     */
+    virtual VectorSpaceVector<Number> &operator*= (const Number factor);
+
+    /**
+     * Divide the entire vector by a fixed factor.
+     */
+    virtual VectorSpaceVector<Number> &operator/= (const Number factor);
+
+    /**
+     * Add the vector @p V to the present one.
+     */
+    virtual VectorSpaceVector<Number> &operator+= (const VectorSpaceVector<Number> &V);
+
+    /**
+     * Substract the vector @p V from the present one.
+     */
+    virtual VectorSpaceVector<Number> &operator-= (const VectorSpaceVector<Number> &V);
+
+    /**
+     * Return the scalar product of two vectors.
+     */
+    virtual Number operator* (const VectorSpaceVector<Number> &V);
+
+    /**
+     * Add @p a to all components. Note that @p a is a scalar not a vector.
+     */
+    virtual void add(const Number a);
+
+    /**
+     * Simple addition of a multiple of a vector, i.e. <tt>*this += a*V</tt>.
+     */
+    virtual void add(const Number a, const VectorSpaceVector<Number> &V);
+
+    /**
+     * Multiple addition of a multiple of a vector, i.e. <tt>*this += a*V+b*W</tt>.
+     */
+    virtual void add(const Number a, const VectorSpaceVector<Number> &V,
+                     const Number b, const VectorSpaceVector<Number> &W);
+
+    /**
+     * Scaling and simple vector addition, i.e. <tt>*this = s*(*this)+V</tt>.
+     */
+    virtual void sadd(const Number s, const VectorSpaceVector<Number> &V);
```

Comment:
This is a more general question: Any reason to have this method? One can easily obtain the desired behavior by `sadd(s, 1.0, V)`. Since we're changing interfaces right now anyway, why not minimize here?